### PR TITLE
META-2730 Create API to attach classifications with dispalyName

### PR DESCRIPTION
**Pull request by nikhilbonte21**
## Change description
https://linear.app/atlanproduct/issue/META-2730/create-api-to-attach-classifications-with-dispalyname-of

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java [hunk 1]
```diff
@@ -1479,20 +1479,28 @@ private void validateEntityAssociations(String guid, List<AtlasClassification> c
...
-            String newClassification = classification.getTypeName();
+        List <AtlasClassification> copyList = new ArrayList<>(classifications);
+        for (AtlasClassification classification : copyList) {
 
-            if (CollectionUtils.isNotEmpty(entityClassifications) && entityClassifications.contains(newClassification)) {
-                throw new AtlasBaseException(AtlasErrorCode.INVALID_PARAMETERS, "entity: " + guid +
-                        ", already associated with classification: " + newClassification);
-            }
+            if (processedTagTypeNames.contains(classification.getTypeName())){
+                classifications.remove(classification);
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:1488]: Why is this required?

Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:1488]: There is flaw in this method.
consider request payload with multiple elements but same combination of "entityGuid" & "displayName"/"typeName"

It this case, as this method checks for associations in graph & passes the validation
Then the same classification gets attached to a same entity multiple times

This change handles this flaw by maintaining processed classification names in the loop

**Review by mehtaanshul — APPROVED** [review]
